admin_solutions/master_node_config.adoc

```diff
@@ -0,0 +1,409 @@
+A cluster architecture of three masters is recommended, and
+link:../architecture/infrastructure_components/kubernetes_infrastructure.html#master[HAproxy] is the recommended solution for this.
+
+[CAUTION]
+====
+If etcd is being installed on the master hosts, then do _not_ configure your
+cluster to use two masters, because etcd will not be able to decide which one
+is authoritative when there are only two cluster members.
+If you are hosting etcd on different hosts, then it is possible to run only two
+masters.
+====
+
+== Configuring Masters and Nodes
+
+The method you use to configure your master and node configuration files must match the method that was used to install your OpenShift cluster. If you followed the:
+
+- link:../install_config/install/advanced_install.html[Advanced installation]
+method using Ansible, then make your configuration changes
+link:../admin_solutions/master_node_config.html#master-node-config-ansible[in the Ansible playbook].
+- link:../install_config/install/quick_install.html[Quick installation]
+ifdef::openshift-origin[]
+or https://docs.openshift.org/latest/getting_started/administrators.html[Manual installation]
+endif::openshift-origin[]
+method, then make your changes
+link:../admin_solutions/master_node_config.html#master-node-config-manual[manually in the configuration files] themselves.
+
+[[master-node-config-ansible]]
+=== Making Configuration Changes Using Ansible
+
+For this section, familiarity with Ansible is assumed, however you can use this
+configuration as a reference to create your own implementation using the
+configuration management tool of your choosing (such as
+https://puppet.com/[Puppet], https://www.chef.io/[Chef],
+http://saltstack.com/[Salt]).
+
+Only a portion of the available options in the configuration files are
+https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example[exposed to Ansible], so
+link:../install_config/install/advanced_install.html#configuring-cluster-variables[some options are configurable]
+with Ansible but not all. After you install OpenShift, Ansible creates an
+inventory file with some values substituted in. Modifying this inventory allows
+you to customize your cluster by affecting changes in the master and node configuration files.
+
+The process of modifying configuration files using Ansible is similar
+to the installation process because it uses the same playbook. When you modify
+the playbook and re-run it, Ansible carries out the required configuration changes.
+
+*Use Case: Configure the cluster to use HTPasswd authentication*
+
+[NOTE]
+====
+This use case assumes you have already set up SSH keys to all the nodes referenced in the playbook.
+====
+
+Initially, the OpenShift cluster would have been installed using a simple command, which you will re-run once you've modified the inventory:
+
+----
+$ ansible-playbook -b -i ./hosts ~/src/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
+----
```

Comment:
I can see the advantages of mentioning the command. But I'm wondering if it's necessary. I feel it might confuse readers at first and they might unnecessarily run the command before they need to.